src/main/java/de/adesso/anki/battle/transfer/BodyDTO.java

```diff
@@ -2,20 +2,67 @@
 
 import de.adesso.anki.battle.util.Position;
 import de.adesso.anki.battle.world.Body;
+import de.adesso.anki.battle.world.DynamicBody;
+import de.adesso.anki.battle.world.bodies.Vehicle;
+import de.adesso.anki.battle.world.bodies.roadpieces.Roadpiece;
+
+
+import de.adesso.anki.sdk.AnkiGateway;
+import de.adesso.anki.sdk.AnkiVehicle;
+import de.adesso.anki.sdk.messages.*;
 
 public class BodyDTO {
 
     private final Body body;
+    
+    private double speed; 
+    
+    private String BTAdress;
 
+    private AnkiVehicle anki; 
+    
+    
+    
+    
+    // find corresponding Anki vehicle
     public BodyDTO(Body body) {
         this.body = body;
+        this.speed = ((DynamicBody)body).getSpeed();
+        if (this.getType().equals("Vehicle")) {
+        	if ( ((Vehicle)body).getAnkiReference() != null) {
+            	this.BTAdress = ((Vehicle)body).getAnkiReference().toString();
+        	}
+        }
+        else {
+        	//Rockets and mines should or in simulation have no BT Adress
+        	this.BTAdress = "NULL";
+        }
+    }
+
+    public double getSpeed() {
+    	return this.speed;
     }
+    
 
+    
     public String getType() {
         return body.getClass().getSimpleName();
     }
 
     public Position getPosition() {
         return body.getPosition();
     }
+ //   public AnkiVehicle getAnkiReference() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/adessoag-github/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.adesso.anki%3Aanki-battle&pullRequest=32&issues=AWSshItHsk71k7lwCc1d&open=AWSshItHsk71k7lwCc1d)

[ISSUE_KEY]: AWSshItHsk71k7lwCc1d